avocado/core/test.py

```diff
@@ -139,7 +139,7 @@ def record_and_warn(*args, **kwargs):
 
         self.debugdir = None
         self.resultsdir = None
-        self.status = None
+        self.status = "ERROR"
```

Comment:
That's the problem I have with "user-facing" statuses and statuses. If we use "NOSTATUS", then we still get the traceback as it's not a user-facing status. For reasons like this I always supported all possible statuses as user-facing statuses.

Btw for example in xunit, test ending with NOSTATUS makes it into the summary, but the test itself is ignored:

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="avocado" tests="1" errors="1" failures="0" skip="0" time="0.000367164611816" timestamp="2016-04-01 07:38:28.096906">
</testsuite>
```

in json and html results it reports as "NOSTATUS", which is not a user-facing result (but I like it):

```
{"tests": [{"status": "NOSTATUS", "url": "passtest.py:PassTest.test", "logfile": "/home/medic/avocado/job-results/job-2016-04-01T07.38-9afec36/test-results/passtest.py:PassTest.test/debug.log", "whiteboard": "", "end": 1459489139.48143, "logdir": "/home/medic/avocado/job-results/job-2016-04-01T07.38-9afec36/test-results/passtest.py:PassTest.test", "start": 1459489139.481079, "test": "passtest.py:PassTest.test", "fail_reason": "None", "time": 0.0003509521484375}], "errors": 1, "job_id": "9afec36c21f44ec77e73bc98371b44887be775e9", "skip": 0, "time": 0.0003509521484375, "debuglog": "/home/medic/avocado/job-results/job-2016-04-01T07.38-9afec36/job.log", "pass": 0, "failures": 0, "total": 1}
```

And human results generate crash:

```
JOB ID     : 7ba814ae6e74a6f9a408e0e5b86abd4f4c3b2c10
JOB LOG    : /home/medic/avocado/job-results/job-2016-04-01T07.43-7ba814a/job.log
TESTS      : 1
 (1/1) passtest.py:PassTest.test:  
Avocado crashed: KeyError: 'NOSTATUS'
Traceback (most recent call last):

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/job.py", line 497, in _run
    replay_map=replay_map)

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/runner.py", line 448, in run_suite
    failures, deadline)

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/runner.py", line 397, in run_test
    self.result.check_test(test_state)

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/result.py", line 110, in check_test
    output_plugin.check_test(state)

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/result.py", line 259, in check_test
    add = status_map[state['status']]

KeyError: 'NOSTATUS'

Please include the traceback info and command line used on your bug report
Report bugs visiting https://github.com/avocado-framework/avocado/issues/new
```
But I see your point. So if we want to keep the user-facing-statuses, how about this change:

```
diff --git a/avocado/core/runner.py b/avocado/core/runner.py
index bf80686..f3eae8e 100644
--- a/avocado/core/runner.py
+++ b/avocado/core/runner.py
@@ -394,6 +394,16 @@ class TestRunner(object):
         if ctrl_c_count > 0:
             self.job.log.debug('')
 
+        if test_state['status'] not in status.user_facing_status:
+            test_state['fail_reason'] = ("Test ended with incorrect test "
+                                         "status %s.\nOriginal fail_reson: %s"
+                                         "\nOriginal fail_class: %s"
+                                         % (test_state.get('status'),
+                                            test_state.get('fail_reason'),
+                                            test_state.get('fail_class')))
+            test_state['fail_class'] = "RUNNER"
+            test_state['status'] = 'ERROR'
+
         self.result.check_test(test_state)
         if not status.mapping[test_state['status']]:
             failures.append(test_state['name'])
```

Normally it does not change anything, but when the status is not in user-facing-mapping it changes it to ERROR and modifies the necessary entries inside status.

@apahim, @clebergnu what do you think? (additionally we can remove all non-human facing statuses from the mapping)